tests/cmake/gen_rst.cmake

```diff
@@ -0,0 +1,73 @@
+find_package(CMinx 1.0.0 PATHS /home/branden/git/CMakeDoc/build NO_DEFAULT_PATH REQUIRED CONFIG)
+include(cminx)
+
+set(cwd "${CMAKE_CURRENT_LIST_DIR}")
+
+
+################################################
+# Defines needed macros for setup and teardown #
+################################################
+
+#[[[
+# Sets needed variables for tests to function.
+#]]
+macro(do_setup)
+  set(input_dir "${cwd}/input")
+  set(output_dir "${cwd}/output")
+  set(input_file "${input_dir}/example.cmake")
+  set(output_file "${output_dir}/example.rst")
+endmacro()
+
+#[[[
+# Removes output directory after each test.
+# Must be manually called, CMakeTest has no
+# built-in teardown functiionality.
+#]]
+macro(do_teardown)
+  file(REMOVE_RECURSE ${output_dir})
+endmacro()
+
+
+##########################
+# Begin test definitions #
+##########################
+
+
+ct_add_test(NAME test_gen_rst)
+function(${test_gen_rst})
+
+  ct_add_section(NAME test_document)
```

Comment:
I now see how the setup/teardown functions could be helpful even with the way sections work. Up to you if you want to add them to CMakeTest.

Reply:
Those functions were mainly ported directly from the Python tests, the `do_setup()` stuff could be done at the top of the root test since all it does is set variables, but the `do_teardown()` needs to be ran after every test which is where the idea for the built-in functionality came from. After this PR and the drafted one in CMakeTest are finished I'll probably add that functionality into CMakeTest.